maat-court-data-api/src/main/java/gov/uk/courtdata/prosecutionconcluded/service/ProsecutionConcludedDAO.java

```diff
@@ -0,0 +1,51 @@
+package gov.uk.courtdata.prosecutionconcluded.service;
+
+import gov.uk.courtdata.entity.RepOrderEntity;
+import gov.uk.courtdata.prosecutionconcluded.dto.ConcludedDTO;
+import gov.uk.courtdata.prosecutionconcluded.helper.CrownCourtCodeHelper;
+import gov.uk.courtdata.prosecutionconcluded.helper.ProcessSentencingHelper;
+import gov.uk.courtdata.prosecutionconcluded.helper.ResultCodeHelper;
+import gov.uk.courtdata.repository.CrownCourtStoredProcedureRepository;
+import gov.uk.courtdata.repository.RepOrderRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class ProsecutionConcludedDAO {
+
+    private final RepOrderRepository repOrderRepository;
+
+    private final CrownCourtStoredProcedureRepository crownCourtStoredProcedureRepository;
+
+    private final CrownCourtCodeHelper crownCourtCodeHelper;
+
+    private final ProcessSentencingHelper processSentencingHelper;
+
+    private final ResultCodeHelper resultCodeHelper;
+
+    public void execute(ConcludedDTO concludedDTO) {
+
+        Integer maatId = concludedDTO.getProsecutionConcluded().getMaatId();
```

Comment:
We should execute this when concluded flag is set to true. The way this has been set now will always trigger the process either with an outcome or null.